create-and-save-map/src/main/java/com/esri/arcgismaps/sample/createandsavemap/screens/MainScreen.kt

```diff
@@ -0,0 +1,462 @@
+    val snackbarHostState = remember { SnackbarHostState() }
+
+    Scaffold(
+        snackbarHost = { SnackbarHost(hostState = snackbarHostState) },
+        topBar = { SampleTopAppBar(title = sampleName) },
+        content = {
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(it),
+                contentAlignment = Alignment.Center
+            ) {
+                var mapScale by remember { mutableIntStateOf(0) }
+                val map = mapViewModel.arcGISMap
+
+                // map view not shown until the portal is loaded and a basemap has been set
+                if (map != null) {
+                    MapView(
+                        modifier = Modifier
+                            .fillMaxSize(),
+                        arcGISMap = map,
+                        // update the map scale in the UI on map scale change
+                        onMapScaleChanged = { currentMapScale ->
+                            if (!currentMapScale.isNaN()) {
+                                mapScale = currentMapScale.roundToInt()
+                            }
+                        },
+                    )
+                }
+
+                val controlsBottomSheetState =
+                    rememberModalBottomSheetState(skipPartiallyExpanded = true)
+                // show the "Edit map" button only when the bottom sheet is not visible
+                if (!controlsBottomSheetState.isVisible) {
+                    ExtendedFloatingActionButton(
+                        modifier = Modifier
+                            .align(Alignment.BottomEnd)
+                            .padding(bottom = 36.dp, end = 24.dp),
+                        onClick = {
+                            composableScope.launch {
+                                controlsBottomSheetState.show()
+                            }
+                        })
+                    {
+                        Icon(Icons.Filled.Edit, contentDescription = "Edit map button")
+                        Spacer(Modifier.padding(8.dp))
+                        Text("Edit Map")
+                    }
+                }
+
+                if (controlsBottomSheetState.isVisible) {
+                    ModalBottomSheet(
+                        modifier = Modifier.wrapContentHeight(),
+                        sheetState = controlsBottomSheetState,
+                        onDismissRequest = {
+                            composableScope.launch {
+                                controlsBottomSheetState.hide()
+                            }
+                        }
+                    ) {
+                        Column(
+                            Modifier
+                                .padding(12.dp)
+                                .navigationBarsPadding()
+                        ) {
+                            BasemapDropdown(
+                                mapViewModel.basemapStyle,
```

Comment:
Can all this Compose code be defined in a separate Composable function/file? Here is an example that I used to move this `Column` into a Composable `CreateMapBottomSheet`: 

```kotlin
CreateMapBottomSheet(
    mapViewModel = mapViewModel,
    onSaveToAccount = {
        composableScope.launch {
            // report success in a snack bar, failure in a popup
            mapViewModel.save().onSuccess {
                snackbarHostState.showSnackbar(
                    message = "Map saved to portal.",
                    withDismissAction = true
                )
            }.onFailure { err ->
                mapViewModel.messageDialogVM.showMessageDialog(
                    title = "Error",
                    description = err.message.toString()
                )
            }
            // dismiss bottom sheet
            isBottomSheetVisible = false
        }
    })
```